pkg/executor/traffic.go

```diff
@@ -80,8 +102,31 @@ type TrafficReplayExec struct {
 // Next implements the Executor Next interface.
 func (e *TrafficReplayExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 	e.Args[startTimeKey] = time.Now().Format(time.RFC3339)
-	form := getForm(e.Args)
-	_, err := request(ctx, e.BaseExecutor, strings.NewReader(form), http.MethodPost, "api/traffic/replay")
+	addrs, err := getTiProxyAddrs(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "get tiproxy addresses failed")
+	}
+	// For shared storage, read the sub-direcotires from the input path and assign each sub-directory to a TiProxy instance.
+	formCtx, cancel := context.WithTimeout(ctx, sharedStorageTimeout)
+	readers, err := formReader4Replay(formCtx, e.Args, len(addrs))
+	cancel()
+	if err != nil {
+		return err
+	}
+	var warning error
+	readerNum, tiproxyNum := len(readers), len(addrs)
+	if readerNum > tiproxyNum {
+		warning = errors.Errorf("tiproxy instances number (%d) is less than input paths number (%d), some paths won't be replayed", tiproxyNum, readerNum)
```

Comment:
In this case, I wonder if it is meaningful to replay the traffic...maybe it's better to report an error?

Reply:
Emm, makes sense.